### ISSUE TITLE
enable spawn gating on Windows

### ISSUE THREAD

**Issue by eladpaz** [frida/frida#2339]
is there any plan to add enable spawn gating (or waiting for spawn) on Windows OS?

**Comment by oleavr** [comment]
Yes, it's something we want to support on all platforms universally. We're still waiting for somebody to volunteer to implement it for Windows though. I don't yet have a use-case for this myself, so I don't know when I'll get around to it. Would you be interested in taking a stab at implementing it?

**Comment by eladpaz** [comment]
I think it is worthwhile presenting my use case. I am using Frida for analyzing malicious scripts and documents, by hooking app (excel, word, cscript) we can observe interesting API calls that can give indication of script or macro intent. my thesis is that by hooking not that many functions we can get a pretty good coverage of most malicious scripts and macros. hopefully this will be a useful tool for other researches as well. something of poor-man's dynamic analysis lab.
however, some malicious activity happens not on the process itself (or children processes), this is the case for example when triggering COM objects such as equation editor (very famous cve-2017-11882 uses buffer overflow in eqnedt32.exe to achieve remote code execution when opening malicious document). in such a case, a separate process will be spawned, outside the scope of our process. 
in short, the use case is - inside malware research lab run frida script, hook interesting api calls and gate additional spawned processes (pre-known).

As much as I am flattered by your proposition, I am not sure I will be of much help.. my C skills are mostly unused for past 10 years..
Thnks for quick reply!
